### PR TITLE
Add functional tests on banner/interstitial

### PR DESCRIPTION
**Pull request by TurpIF**
# Tested scenario
### Display a banner ad
- Given already prefetch valid bid
- Given MoPub response pointing to Criteo adapter
- When loading the MoPub banner
- And waiting for idle state
- Then MoPub listener is notified for success
- Then MoPub view contains the display URL in a JS script tag

### No bid is available
- Given already prefetch invalid bid (or just no bid)
- Given MoPub response pointing to Criteo adapter
- When loading the MoPub banner
- And waiting for idle state
- Then MoPub listener is notified for failure

### Click on ad redirects to banner page
- Given already prefetch valid bid
- Given MoPub response pointing to Criteo adapter
- When loading the MoPub banner
- And waiting for idle state
- And clicking on banner
- Then user is redirected to banner URI
- Then MoPub listener is notified for click

### Loading and showing an interstitial ad
- Given already prefetch valid bid
- Given MoPub response pointing to Criteo adapter
- When loading the MoPub interstitial
- And waiting for idle state
- And showing the interstitial
- And waiting for an activity to be opened
- Then MoPub listener is notified for success
- Then MoPub listener is notified for an shown interstitial
- Then interstitial contains a web view with the ad creative

### No bid is available
- Given already prefetch invalid bid
- Given MoPub response pointing to Criteo adapter
- When loading the MoPub interstitial
- And waiting for idle state
- Then MoPub listener is notified for failure

### Click on ad redirects to interstitial page
- Given already prefetch valid bid
- Given MoPub response pointing to Criteo adapter
- When loading the MoPub interstitial
- And waiting for idle state
- And showing the interstitial
- And clicking on it
- Then user is redirected to interstitial URI
- Then MoPub listener is notified for click

### CODE REVIEW

File: mediation/src/androidTest/java/com/criteo/mediation/mopub/MoPubHelper.java [hunk 1]
```diff
@@ -21,6 +21,13 @@
 
 public class MoPubHelper {
 
+  // /!\ Do not replace those class names with programmatic expressions. The class are specified on
+  // MoPub server by each publisher for each of their ad units. A renaming of our class should make
```
Comment (Amokrane) [inline mediation/src/androidTest/java/com/criteo/mediation/mopub/MoPubHelper.java:25]: Renaming any of these classes must make the tests fail

File: mediation/src/androidTest/java/com/criteo/publisher/view/WebViewClickerExt.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        isHtmlLoaded.countDown()
+      }
+    })
+  }
+
+  isHtmlLoaded.await()
+}
+
+fun WebView.simulateClickOnAd() {
+  // TODO kind of duplication of [WebViewClicker#simulateClickOnAd]
```
Comment (Amokrane) [inline mediation/src/androidTest/java/com/criteo/publisher/view/WebViewClickerExt.kt:24]: Do you plan to address the TODO in a separate commit of the same PR or later?

File: mediation/src/androidTest/java/com/criteo/publisher/view/WebViewLookupExt.kt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.criteo.publisher.view
+
+import android.view.View
+import java.util.concurrent.Future
+
+fun WebViewLookup.lookForNonEmptyHtmlContent(view: View): Future<String> {
+  // TODO move this in test-utils module
```
Comment (Amokrane) [inline mediation/src/androidTest/java/com/criteo/publisher/view/WebViewLookupExt.kt:7]: Ditto about the TODO.

File: mediation/src/androidTest/java/com/criteo/publisher/view/WebViewClickerExt.kt (hunk 2 above)
Comment (TurpIF) [reply mediation/src/androidTest/java/com/criteo/publisher/view/WebViewClickerExt.kt:24]: This change require update of the test-utils module in the SDK repository + waiting for the snapshot deployment.
This TODO is something we can live with, so we can address in after.

File: mediation/src/androidTest/java/com/criteo/publisher/view/WebViewLookupExt.kt (hunk 3 above)
Comment (TurpIF) [reply mediation/src/androidTest/java/com/criteo/publisher/view/WebViewLookupExt.kt:7]: Same than above.

File: mediation/src/androidTest/java/com/criteo/mediation/mopub/MoPubHelper.java (hunk 1 above)
Comment (TurpIF) [reply mediation/src/androidTest/java/com/criteo/mediation/mopub/MoPubHelper.java]: Done

**Pull request merged**